src/java/org/apache/cassandra/db/marshal/AbstractCompositeType.java

```diff
@@ -316,6 +320,64 @@ public  <V> void validate(V input, ValueAccessor<V> accessor)
         }
     }
 
+    // a int, b text
+    // PARTITION KEY ((a, b), c)
+
+    @Override
+    public void checkConstraints(ByteBuffer input, ColumnConstraints constraints) throws ConstraintViolationException
+    {
+        boolean allNoopConstraint = true;
+        for (ColumnConstraint c : constraints.getConstraints())
+        {
+            assert c instanceof ColumnConstraints;
+            allNoopConstraint = allNoopConstraint && c instanceof ColumnConstraints.Noop;
```

Comment:
as soon as `allNoopConstraint` is `false`, you can `break` this for-loop.

Reply:
@bbotella isn't there a bug? Should not it be `if (!allNoopConstraint) return` ?